### PR TITLE
Refactor python wheels CI and use Github Cache

### PR DESCRIPTION
**Pull request by mwestphal**
- Refactor python wheels CI into an action
 - Use action in nightly and release  workflow
 - Use github cache to store f3d-superbuild install dir

### CODE REVIEW

File: .github/actions/wheels/action.yml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+      fetch-depth: 0
+      path: 'f3d'
+
+  - name: Recover VTK version to build
+    working-directory: ${{github.workspace}}
+    shell: bash
+    run: |
+      echo VTK_COMMIT_SHA=`cat ./f3d/.github/actions/vtk_commit_sha` >> $GITHUB_ENV
+
+  - name: fsbb directory
```
Comment (Meakk) [inline .github/actions/wheels/action.yml:28]: what means `fsbb`?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -46,3 +46,25 @@ jobs:
...
+        os: [ubuntu, windows, macos]
+
+    runs-on: ${{matrix.os}}-latest
+
+    env:
+      DISPLAY: :0
+
+    steps:
+
+    - name: Checkout
```
Comment (Meakk) [inline .github/workflows/ci.yml:63]: Any reason you are not putting this step in the action?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -69,3 +69,47 @@ jobs:
...
+        prerelease: true
+        allowUpdates: true
+        updateOnlyUnreleased: true
+        replacesArtifacts: true
+        omitBodyDuringUpdate: true
+        omitNameDuringUpdate: true
+        tag: ${{github.event.inputs.f3d_version}}
+        artifacts: './wheelhouse/f3d-*.whl'
+
+#TODO publich to pypi
```
Comment (Meakk) [inline .github/workflows/release.yml:115]:
```suggestion
#TODO publish to pypi
```

**Review by Meakk — COMMENTED** [review]
A few minor comments but I like it, thanks!

File: .github/actions/wheels/action.yml (hunk 1 above)
Comment (mwestphal) [reply .github/actions/wheels/action.yml:28]: **F**3D **S**uperbuild **b**uild

File: .github/workflows/ci.yml (hunk 2 above)
Comment (mwestphal) [reply .github/workflows/ci.yml:63]: you need the code of an action to use it, see the next step

Comment (Meakk) [reply .github/workflows/ci.yml:63]: Yeah, obviously, thanks!

**Review by Meakk — APPROVED** [review]

**Review by Meakk — APPROVED** [review]

**Pull request merged**